src/resources/reference-data/ReferenceDataForm.tsx

```diff
@@ -201,13 +211,29 @@ const FormContent = ({
       setIsIDChanged(true)
     }
   }, [{ ...dirtyFields }])
+  const validName = name ?? ''
+  const resourceForEditId = [
+    constants.R_CAT_CODE,
+    constants.R_CAT_CAVE,
+    constants.R_CAT_HANDLE,
+    constants.R_DEPARTMENT
+  ]
 
-  const warningTextForId =
-    'Warning: Editing the id of a Department that is in use may lead to data corruption.  The id of a department must not be modified if data has been assigned to that department.'
+  const resourceName =
+    name === constants.R_DEPARTMENT
+      ? 'department'
+      : name === constants.R_CAT_CODE
```

Comment:
The strings for `code`, `cave` and `handle` have to come from the `ConfigData`, please.

Reply:
Oky let me check.

Reply:
Done.